new-log-viewer/src/App.tsx

```diff
@@ -10,9 +11,11 @@ import StateContextProvider from "./contexts/StateContextProvider";
 const App = () => {
     return (
         <>
-            <StateContextProvider>
-                <Layout/>
-            </StateContextProvider>
+            <UrlContextProvider>
+                <StateContextProvider>
```

Comment:
Should we document that `StateContextProvider` depends on `UrlContextProvider`? I'm not sure what the convention is in projects that use layers of context providers.

Reply:
Sure. Maybe we can discuss more about their relationship before we document.

As we previously discussed, at the end of the day we do have the plan to provide the Log Viewer as a React component to other apps, which accepts certain properties parameters for external state control.
One way to do that is to let `UrlContextProvider` accept those properties and provide those without any url handling. Do you have any idea how we can bypass the `UrlContextProvider`?

Reply:
Haven't thought about it in detail, but perhaps we could do something like:
* Make the UrlContextProvider more general such that it essentially maintains global storage for the parameters important to the log viewer (`filePath` and `logEventNum` right now) and provides methods for updating those, but it doesn't do URL manipulation by default.
* Make the provider take in methods that get called whenever the parameters are updated, that way the URL-manipulation can be external to the core log viewer and apps that don't want to do URL-manipulation can provide their own controls.

Reply:
The first approach sounds more straight forward. Shall we rename the FC as `ParamsContextProvider`?

Reply:
Updated `StateContextProvider` doc for its dependency on `UrlContextProvider`.